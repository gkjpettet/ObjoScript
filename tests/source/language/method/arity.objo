class Foo {
  method() { return "no args" }
  method(a) {return  a }
  method(a, b) { return a + b }
  method(a, b, c) { return a + b + c }
  method(a, b, c, d) { return a + b + c + d }
  method(a, b, c, d, e) { return a + b + c + d + e }
  method(a, b, c, d, e, f) { return a + b + c + d + e + f }
  method(a, b, c, d, e, f, g) { return a + b + c + d + e + f + g }
  method(a, b, c, d, e, f, g, h) { return a + b + c + d + e + f + g + h }
  method(a, b, c, d, e, f, g, h, i) { return a + b + c + d + e + f + g + h + i }
  method(a, b, c, d, e, f, g, h, i, j) { return a + b + c + d + e + f + g + h + i + j }
  method(a, b, c, d, e, f, g, h, i, j, k) { return  a + b + c + d + e + f + g + h + i + j + k }
  method(a, b, c, d, e, f, g, h, i, j, k, l) { return a + b + c + d + e + f + g + h + i + j + k + l }
  method(a, b, c, d, e, f, g, h, i, j, k, l, m) { return  a + b + c + d + e + f + g + h + i + j + k + l + m }
  method(a, b, c, d, e, f, g, h, i, j, k, l, m, n) { return  a + b + c + d + e + f + g + h + i + j + k + l + m + n }
  method(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o) { return  a + b + c + d + e + f + g + h + i + j + k + l + m + n + o }
  method(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p) { return  a + b + c + d + e + f + g + h + i + j + k + l + m + n + o + p }
}

var foo = Foo()
System.print(foo.method()) // expect: no args
System.print(foo.method(1)) // expect: 1
System.print(foo.method(1, 2)) // expect: 3
System.print(foo.method(1, 2, 3)) // expect: 6
System.print(foo.method(1, 2, 3, 4)) // expect: 10
System.print(foo.method(1, 2, 3, 4, 5)) // expect: 15
System.print(foo.method(1, 2, 3, 4, 5, 6)) // expect: 21
System.print(foo.method(1, 2, 3, 4, 5, 6, 7)) // expect: 28
System.print(foo.method(1, 2, 3, 4, 5, 6, 7, 8)) // expect: 36
System.print(foo.method(1, 2, 3, 4, 5, 6, 7, 8, 9)) // expect: 45
System.print(foo.method(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)) // expect: 55
System.print(foo.method(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11)) // expect: 66
System.print(foo.method(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12)) // expect: 78
System.print(foo.method(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13)) // expect: 91
System.print(foo.method(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14)) // expect: 105
System.print(foo.method(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15)) // expect: 120
System.print(foo.method(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16)) // expect: 136