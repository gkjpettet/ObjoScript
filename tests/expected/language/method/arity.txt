no args
1
3
6
10
15
21
28
36
45
55
66
78
91
105
120
136